src/main/java/uk/gov/hmcts/darts/authentication/config/AuthenticationConfiguration.java

```diff
@@ -1,58 +1,16 @@
 package uk.gov.hmcts.darts.authentication.config;
 
-import lombok.Getter;
-import lombok.extern.slf4j.Slf4j;
-import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
+import uk.gov.hmcts.darts.authentication.exception.AuthenticationException;
 
-@Slf4j
 @Configuration
-@Getter
 public class AuthenticationConfiguration {
 
-    @Value("${spring.security.oauth2.client.registration.external-azure-ad.client-id}")
-    private String externalADclientId;
-
-    @Value("${spring.security.oauth2.client.registration.external-azure-ad.client-secret}")
-    private String externalADclientSecret;
-
-    @Value("${spring.security.oauth2.client.provider.external-azure-ad-provider.authorization-uri}")
-    private String externalADauthorizationUri;
-
-    @Value("${spring.security.oauth2.client.provider.external-azure-ad-provider.logout-uri}")
-    private String externalADlogoutUri;
-
-    @Value("${spring.security.oauth2.client.provider.external-azure-ad-provider.token-uri}")
-    private String externalADtokenUri;
-
-    @Value("${spring.security.oauth2.client.provider.external-azure-ad-provider.jwk-set-uri}")
-    private String externalADjwkSetUri;
-
-    @Value("${spring.security.oauth2.client.registration.external-azure-ad.scope}")
-    private String externalADscope;
-
-    @Value("${spring.security.oauth2.client.registration.external-azure-ad.redirect-uri}")
-    private String externalADredirectUri;
-
-    @Value("${spring.security.oauth2.client.registration.external-azure-ad.logout-redirect-uri}")
-    private String externalADlogoutRedirectUri;
-
-    @Value("${spring.security.oauth2.client.registration.external-azure-ad.authorization-grant-type}")
-    private String externalADauthorizationGrantType;
-
-    @Value("${spring.security.oauth2.client.registration.external-azure-ad.response-type}")
-    private String externalADresponseType;
-
-    @Value("${spring.security.oauth2.client.registration.external-azure-ad.response-mode}")
-    private String externalADresponseMode;
-
-    @Value("${spring.security.oauth2.client.registration.external-azure-ad.prompt}")
-    private String externalADprompt;
-
-    @Value("${spring.security.oauth2.client.registration.external-azure-ad.issuer-uri}")
-    private String externalADissuerUri;
-
-    @Value("${spring.security.oauth2.client.provider.external-azure-ad-provider.reset-password-uri}")
-    private String externalADresetPasswordUri;
-
+    @Bean
+    public AuthConfigFallback getDefaultNoAuthConfiguration() {
+        return (req) -> {
+            throw new AuthenticationException("No authentication configuration found");
```

Comment:
Preferred error-handling pattern is to throw a `DartsApiException` with the appropriate enum value as a param.

Can you replace `throw new AuthenticationException` with `throw new DartsApiException(AuthenticationError.FAILED_TO_OBTAIN_AUTHENTICATION_CONFIG)`, and add the new enum value to `src/main/java/uk/gov/hmcts/darts/authentication/exception/AuthenticationError.java` along with a unique error code?

Would also be worth removing (or marking as deprecated) the `AuthenticationException` class as this should no longer be used.

Reply:
Done